admin_auto_filters/filters.py

```diff
@@ -1,35 +1,118 @@
-from django.contrib.admin.widgets import AutocompleteSelect as Base
-from django import forms
-from django.contrib import admin
+"""Defines autocomplete filters and helper functions for the admin."""
+
+import warnings
+from django.contrib.admin import SimpleListFilter
+from django.contrib.admin.utils import prepare_lookup_value
+from django.contrib.admin.widgets import (
+    AutocompleteSelect as BaseAutocompleteSelect,
+    AutocompleteSelectMultiple as BaseAutocompleteSelectMultiple,
+)
+from django.core.exceptions import ImproperlyConfigured
 from django.db.models import ForeignObjectRel
 from django.db.models.constants import LOOKUP_SEP  # this is '__'
-from django.db.models.fields.related_descriptors import ReverseManyToOneDescriptor, ManyToManyDescriptor
+from django.db.models.fields.related_descriptors import (
+    ReverseManyToOneDescriptor, ManyToManyDescriptor,
+)
+from django.forms import ModelChoiceField, ModelMultipleChoiceField
 from django.forms.widgets import Media, MEDIA_TYPES, media_property
 from django.shortcuts import reverse
 
 
-class AutocompleteSelect(Base):
+class AutocompleteSelect(BaseAutocompleteSelect):
+    """A customize AutocompleteSelect that allows a custom URL."""
+
     def __init__(self, rel, admin_site, attrs=None, choices=(), using=None, custom_url=None):
+        """Initialize class variables for an AutocompleteSelect object."""
         self.custom_url = custom_url
         super().__init__(rel, admin_site, attrs, choices, using)
     
     def get_url(self):
+        """Specifies the URL to be used to fetch the autocomplete list."""
+        return self.custom_url if self.custom_url else super().get_url()
+
+
+class AutocompleteSelectMultiple(BaseAutocompleteSelectMultiple):
```

Comment:
You are repeating yourself with these two select classes. Wouldn't it be better to create one base class and inherit from it?
```
class AutocompleteSelectWidget:
    def __init__(self, rel, admin_site, attrs=None, choices=(), using=None, custom_url=None):
        self.custom_url = custom_url
        super().__init__(rel, admin_site, attrs, choices, using)

    def get_url(self):
        return self.custom_url if self.custom_url else super().get_url()

class AutocompleteSelect(AutocompleteSelectWidget, BaseAutocompleteSelect):
    pass

class AutocompleteSelectMultiple(AutocompleteSelectWidget, BaseAutocompleteSelectMultiple):
    pass
```
I would change those names btw. - creating the name which is the same as the name of class in django widget is confusing for me, but it's your call